test/extended/cpu_partitioning/crio.go

```diff
@@ -0,0 +1,314 @@
+package cpu_partitioning
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	ocpv1 "github.com/openshift/api/config/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+)
+
+// This collection script queries CRIO for all it's containers, that data is then filtered down to only a subset
+// of information needed for validating workload partitioning and the PIDs are mapped to their corresponding host
+// CPUSet via the taskset command.
+const collectionScript = `
+#!/bin/bash
+
+container_ids=$(sudo crictl ps -q)
+container_data=$(sudo crictl inspect $container_ids)
+
+workload_containers=$(echo $container_data | jq -rs '[ 
+	.[] | select(.info.runtimeSpec.annotations["target.workload.openshift.io/management"]) | 
+	{
+		containerCPUSet: (.info.runtimeSpec.linux.resources.cpu.cpus // ""),
+        containerCPUShare: .info.runtimeSpec.linux.resources.cpu.shares,
+		containerAnnotations: .info.runtimeSpec.annotations,
+        containerPodNamespace: .info.runtimeSpec.annotations["io.kubernetes.pod.namespace"],
+        containerPodName: .info.runtimeSpec.annotations["io.kubernetes.pod.name"],
+		containerName: .status.metadata.name,
+		pid: .info.pid, 
+		hostname: .info.runtimeSpec.hostname 
+	}
+	]')
+
+workload_host_cpu_set=$(echo $workload_containers | jq -r '.[]|.pid' | \
+while read i; do taskset -cp $i; done | awk '{printf "{\"hostCPUSet\": \"%s\"}", $6}' | jq -rs)
+
+final_results=$(echo "$workload_containers $workload_host_cpu_set" | jq -rs '.[0] as $file1 | .[1] as $file2 | map($file1[] + $file2[]) | unique')
+
+echo "$final_results"
+`
+
+// Collection data for each container in a node
+type crioContainerData struct {
+	ContainerPodNamespace string            `json:"containerPodNamespace"`
```

Comment:
nit: I would remove the `Container` prefix from the variables names, since the struct name is `crioContainerData` it's clearly you're referring to container's data.
This would made the code cleaner and shorter.

Reply:
good suggestion, changed